redun/executors/aws_utils.py

```diff
@@ -59,6 +61,38 @@ def get_aws_client(service: str, aws_region: str = DEFAULT_AWS_REGION) -> boto3.
     return client
 
 
+def is_ec2_instance() -> bool:
+    """
+    Returns True if this process is running on an EC2 instance.
+
+    We use the presence of a link-local address as a sign we are on an EC2 instance.
+    """
+    try:
+        resp = urlopen("http://169.254.169.254/latest/meta-data/", timeout=1)
+        return resp.status == 200
+    except URLError:
+        return False
+
+
+def get_aws_env_vars() -> Dict[str, str]:
```

Comment:
I generalized this function from `aws_batch.py`. This function helps with propagating AWS credentials from the scheduler's host into job Docker containers.